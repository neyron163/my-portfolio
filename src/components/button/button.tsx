import classNames from "classnames";
import React from "react";

import s from "./button.module.scss";

export interface PropsType {
  children: JSX.Element | string | number;
  href?: string;
  width: "small" | "medium" | "long";
  className?: string;
}

export const Button = ({
  children,
  href,
  width,
  className,
}: PropsType): React.ReactElement => {
  if (href) {
    return (
      <a href={href} className={classNames(s.button, s.link, s[width], className)}>
        {children}
      </a>
    );
  }
  return <button className={classNames(s.button, s.link, s[width])}>{children}</button>;
};

Button.defaultProps = {
  width: "small",
};
