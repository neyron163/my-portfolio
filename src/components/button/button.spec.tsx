import React from 'react';
import renderer from 'react-test-renderer';

import {Button, PropsType} from './button';

let props: PropsType;

beforeEach(() => {
    props = {
        children: 'child',
        width: 'small',
    }
})

describe('<Button />', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<Button> children </Button>);
        expect(tree).toMatchSnapshot();
    });

    it('should render correctly if prop href exist', () => {
        props.href = 'link';
        const tree = renderer.create(<Button {...props} />);
        expect(tree).toMatchSnapshot();
    });

    it('should render correctly if width medium', () => {
        props.width = 'medium';
        const tree = renderer.create(<Button {...props} />);
        expect(tree).toMatchSnapshot();
    });

    it('should render correctly if width long', () => {
        props.width = 'long';
        const tree = renderer.create(<Button {...props} />);
        expect(tree).toMatchSnapshot();
    });
});