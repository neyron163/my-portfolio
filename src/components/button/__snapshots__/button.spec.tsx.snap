// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
<button
  className="button link small"
>
   children 
</button>
`;

exports[`<Button /> should render correctly if prop href exist 1`] = `
<a
  className="button link small"
  href="link"
>
  child
</a>
`;

exports[`<Button /> should render correctly if width long 1`] = `
<button
  className="button link long"
>
  child
</button>
`;

exports[`<Button /> should render correctly if width medium 1`] = `
<button
  className="button link medium"
>
  child
</button>
`;
